---
title: "Midterm Analysis"
output: html_document
date: "2024-10-24"
author: Samantha Zocher
---
## Step 1 - Read in data into R
```{r}
#Read in the clinical data and view 
clinData <- read.csv(file = "input/Blalock_clin_final.csv",
                       row.names = 1)
knitr::kable(head(clinData[,1:4],n = 5))
```
```{r}
#Read in gene expression data 
geneExp <- read.table(file = "input/GSE62232_Blalock_geneexp_final.tsv",
                      row.names = 1,
                       header = T,
                       stringsAsFactors = F)
#View it in a table
knitr::kable(geneExp[1:5, 1:4])
```
## Step 2 - Clean/Filter data
### Filter clinical data - not needed
Clinical data has one row per patient.
### Filter gene exp data
Check rows and columns:

-   Rows are features (genes in this case)
-   Columns are patients


Now we can connect the data using the Biospeciman ID

```{r}
matchingSamples <- which(colnames(geneExp) %in% clinData$BIOSPECIMEN_ID) # all 30 IDs matching
subsetGeneExp <- geneExp[, matchingSamples] 
```
## Step 3 - Identify the groups to be compared
Now we will compare the moderately affected group to the incipient group. 
Baseline will be the incipient group
Comparison will be the moderately affected group
```{r}
# Labels (row numbers) that can identify the baseline group patients
baselineGrpLabels <-  which(clinData$DISEASE_STATUS == "Incipient") # 7 samples
head(baselineGrpLabels)
# Use the labels (row numbers) to subset baseline patients in clinical data file 
clinBase <- clinData[baselineGrpLabels, ]


# Labels (row numbers) that can identify the comp group patients
compGrpLabels <- which(clinData$DISEASE_STATUS == "Moderate") #8 samples
head(compGrpLabels)
# Use the labels (row numbers) to subset comp patients in clinical data file 
clinComp <- clinData[compGrpLabels, ]


#### Use the clinBase and clinComp objects to subset gene expression data
geneExpBase1 <- subsetGeneExp[, clinBase$BIOSPECIMEN_ID] # 43135 feature (rows), 7 samples columns
geneExpComp1 <- subsetGeneExp[, clinComp$BIOSPECIMEN_ID] # 43135 feature (rows), 8 samples columns
```

## Step 4: Sanity check
```{r}
#See if sample ids in clinical data match sample ids in gene exp data
clinBase$BIOSPECIMEN_ID == colnames(geneExpBase1)
clinComp$BIOSPECIMEN_ID == colnames(geneExpComp1)

### Checking if the samples IDs baseline and comp groups are the same
write.table(clinBase$BIOSPECIMEN_ID , file = "output/ClinBaseIDs.tsv", sep="\t", quote = F  )
write.table(clinComp$BIOSPECIMEN_ID , file = "output/ClinCompIDs.tsv", sep="\t", quote = F  )
write.table(colnames(geneExpBase1) , 
            file = "output/GeneExpBaseIDs.tsv", 
            sep="\t", quote = F  )
write.table(colnames(geneExpComp1) , 
            file = "output/GeneExpCompIDs.tsv", 
            sep="\t", quote = F  )

#Gene expression data 
write.table(rownames(geneExp),file = "output/genExpFeatureIDsCheck.tsv", sep="\t", quote = F  )
#I have investigated these in excel and the data has passed the sanity check
```
## Step 5: Prep data for T-test

```{r}
### Checking to make sure data is a numeric data frame
knitr::kable(head(geneExpBase1[1:5,1:4]))
knitr::kable(head(geneExpComp1[1:5,1:4]))

#Loading the t test function
source("fnTTest.R")
```

```{r}
# Call T-test function
results1 = fnTTest(baseGroup = geneExpBase1, 
                   compGroup = geneExpComp1, 
                   testName = "TTest_", 
                   baseGroupName = "Incipient",
                   compGroupName =  "Moderate",
                   folderName = "output")
```


### Sub-set top differentially expressed genes


```{r}
#Read in the T-Test results file
ttestResults1 <- read.csv(file = "output/TTest__Moderate_(Comp).vs._Incipient_(Base).TTest.csv")

#sort by p-value
ttestResultsSorted1 <- dplyr::arrange(ttestResults1, Pvalue)

#find rows with p-value < 0.01
whichSig <- which(ttestResultsSorted1$Pvalue <= 0.01) 

#Short list sig results
ttestResultsSig1 <- ttestResultsSorted1[whichSig, ] 


### Export short listed results
write.table(x = ttestResultsSig1, 
            file = "output/Ttest1_Shortlisted.tsv", 
            quote = F, sep = "\t")
```

```{r}
funcSplit <- function(featureX) {
  f1 <- unlist(strsplit(x = featureX, split = "|", fixed = TRUE))
  f2 <- f1[2]
  return(f2)
}
geneNames1 <- apply(X = as.matrix(ttestResultsSig1$Feature), 
                    MARGIN = 1, FUN = funcSplit)

head(geneNames1)

#print length of short listed gene names
length(geneNames1)

### Export list of gene names
write.table(x = geneNames1, 
            file = "output/SigDiffExpressedGenes1.tsv", 
            quote = F, sep = "\t")
```
# This is the end of our first comparison: incipient vs moderate. Now we will begin the analysis for severely affected vs moderately affected 

Since we already have the clinical and genetic data loaded into our environment, we can begin from step 3

## Step 3 - Identify the groups to be compared
Our baseline group will be moderately affected patients
Our comparison group will be the severely affected patients. 

```{r}
# Labels (row numbers) that can identify the baseline group patients
baselineGrpLabels2 <-  which(clinData$DISEASE_STATUS == "Moderate") #8 samples
head(baselineGrpLabels2)
# Use the labels (row numbers) to subset baseline patients in clinical data file 
clinBase2 <- clinData[baselineGrpLabels2, ]

# Labels (row numbers) that can identify the comp group patients
compGrpLabels2 <- which(clinData$DISEASE_STATUS == "Severe") #7 samples
head(compGrpLabels2)
# Use the labels (row numbers) to subset comp patients in clinical data file 
clinComp2 <- clinData[compGrpLabels2, ]

#### Use the clinBase and clinComp objects to subset gene expression data
geneExpBase2 <- subsetGeneExp[, clinBase2$BIOSPECIMEN_ID] # 43135 feature (rows), 8 samples columns
geneExpComp2 <- subsetGeneExp[, clinComp2$BIOSPECIMEN_ID] # 43135 feature (rows), 7 samples columns
```
# Sanity check 

```{r}
#See if sample ids in clinical data match sample ids in gene exp data
clinBase2$BIOSPECIMEN_ID == colnames(geneExpBase2)
clinComp2$BIOSPECIMEN_ID == colnames(geneExpComp2)

### Checking if the samples IDs baseline and comp groups are the same
write.table(clinBase2$BIOSPECIMEN_ID , file = "output/ClinBaseIDs2.tsv", sep="\t", quote = F  )
write.table(clinComp2$BIOSPECIMEN_ID , file = "output/ClinCompIDs2.tsv", sep="\t", quote = F  )
write.table(colnames(geneExpBase2) , 
            file = "output/GeneExpBaseIDs2.tsv", 
            sep="\t", quote = F  )
write.table(colnames(geneExpComp2) , 
            file = "output/GeneExpCompIDs2.tsv", 
            sep="\t", quote = F  )

#I have investigated these in excel and the data has passed the sanity check
```

## Step 5: Prep for t-test
```{r}
### Checking to make sure data is a numeric data frame
knitr::kable(head(geneExpBase2[1:5,1:4]))
knitr::kable(head(geneExpComp2[1:5,1:4]))
```
```{r}
# Perform t test
results2 = fnTTest(baseGroup = geneExpBase2, 
                   compGroup = geneExpComp2, 
                   testName = "TTest_", 
                   baseGroupName = "Moderate",
                   compGroupName =  "Severe",
                   folderName = "output")
```

### Subset differentially expressed genes
```{r}
#Read in the T-Test results file
ttestResults2 <- read.csv(file = "output/TTest__Severe_(Comp).vs._Moderate_(Base).TTest.csv")

#sort by p-value
ttestResultsSorted2 <- dplyr::arrange(ttestResults2, Pvalue)

#find rows with p-value < 0.01
whichSig <- which(ttestResultsSorted2$Pvalue <= 0.01) 

#Short list sig results
ttestResultsSig2 <- ttestResultsSorted2[whichSig, ] 


### Export short listed results
write.table(x = ttestResultsSig2, 
            file = "output/Ttest1_Shortlisted2.tsv", 
            quote = F, sep = "\t")
```

Make a data frame with just the gene names
```{r}
geneNames2 <- apply(X = as.matrix(ttestResultsSig2$Feature), 
                    MARGIN = 1, FUN = funcSplit)

head(geneNames2)

#print length of short listed gene names
length(geneNames2)

### Export list of gene names
write.table(x = geneNames2, 
            file = "output/SigDiffExpressedGenes2.tsv", 
            quote = F, sep = "\t")
```
This concludes our 2 comparisons. View the separate .rmd file for the Enrichment analysis. 
